### PR TITLE
Replace Assert annotations with attributes, remove Assert annotation configuration block

### PR DESCRIPTION
**Pull request by ker0x**
Following #17155, this one replace all Assert annotations!

### CODE REVIEW

**Review by mdoutreluingne — APPROVED** [review]

File: reference/constraints/Collection.rst [hunk 1]
```diff
@@ -223,14 +193,18 @@ you can do the following:
...
+                fields: [
+                    'personal_email' => new Assert\Required([
+                        new Assert\NotBlank,
+                        new Assert\Email,
+                    ]),
+                    'alternate_email' => new Assert\Optional(
+                        new Assert\Email
+                    ),
+                ],
+                allowMissingFields: true,
```
Comment (xabbuh) [inline reference/constraints/Collection.rst]: Is this required?

Comment (ker0x) [reply reference/constraints/Collection.rst]: No indeed, wrong copy/paste

File: reference/constraints/Collection.rst [hunk 2]
```diff
@@ -214,7 +184,7 @@ you can do the following:
 
 .. configuration-block::
 
-    .. code-block:: php-annotations
+    .. code-block:: php-attributes
```
Comment (wouterj) [inline reference/constraints/Collection.rst:187]: Ideally, we should backport the addition of this code block to 5.4

**Review by wouterj — APPROVED** [review]

**Comment by wouterj** [comment]
Wow, thanks for all your efforts on this annotation/attribute topic Romain!

File: reference/constraints/Collection.rst (hunk 2 above)
Comment (xabbuh) [reply reference/constraints/Collection.rst:187]: see #17219

**Pull request merged**